### ISSUE TITLE
CCP asserts if conditional branch is done on a null value

### ISSUE THREAD

**Issue by alan-baker** [KhronosGroup/SPIRV-Tools#1338]
VisitBranchCondition doesn't handle the case of a boolean null value.

**Comment by alan-baker** [comment]
```
OpCapability Shader
OpMemoryModel Logical GLSL450
OpEntryPoint Fragment %func "func"
%void = OpTypeVoid
%bool = OpTypeBool
%int = OpTypeInt 32 1
%null = OpConstantNull %bool
%int_1 = OpConstant %int 1
%int_2 = OpConstant %int 2
%functy = OpTypeFunction %void
%func = OpFunction %void None %functy
%1 = OpLabel
OpSelectionMerge %2 None
OpBranchConditional %null %2 %3
%3 = OpLabel
OpBranch %2
%2 = OpLabel
%phi = OpPhi %int %int_1 %1 %int_2 %2
%add = OpIAdd %int %int_1 %phi
```

The above testcase asserts in CCP.

**Comment by alan-baker** [comment]
Similar issue for switches also fixed by #1339
